src/components/calculator.js

```diff
@@ -0,0 +1,103 @@
+import { PropTypes } from 'prop-types';
+import { useState } from 'react';
+import calculate from '../logic/calculate';
+import Quotes from './quotes';
+
+const ans = document.querySelector('.answer');
+
+const Calculator = () => {
+  const [totalVal, setTotalVal] = useState(null);
+  const [nextVal, setNextVal] = useState(null);
+  const [operationVal, setOperationVal] = useState(null);
+
+  const onClickHandler = (e) => {
+    const res = calculate({
+      total: totalVal,
+      next: nextVal,
+      operation: operationVal,
+    }, e.target.innerHTML);
+    const ans = document.querySelector('.answer');
+    setTotalVal(res.total);
+    setNextVal(res.next);
+    setOperationVal(res.operation);
+    ans.innerHTML = (!res.next) ? res.total : res.next;
+  };
+
+  const onHandleKeyDown = () => {
+    ans.innerHTML = 'E';
+  };
+
+  return (
+    <>
+      <div className="calculator">
+        <div className="grid-container">
+          <div className="grid-item" id="answer-div">
+            <Result val="0" />
+          </div>
+          <Item val="AC" onClick={onClickHandler} />
+          <Item val="+/-" onClick={onClickHandler} />
+          <Item val="%" onClick={onClickHandler} />
+          <Operators val="÷" onClick={onClickHandler} />
+          <Item val="7" onClick={onClickHandler} />
+          <Item val="8" onClick={onClickHandler} />
+          <Item val="9" onClick={onClickHandler} />
+          <Operators val="x" onClick={onClickHandler} />
+          <Item val="4" onClick={onClickHandler} />
+          <Item val="5" onClick={onClickHandler} />
+          <Item val="6" onClick={onClickHandler} />
+          <Operators val="-" onClick={onClickHandler} />
+          <Item val="1" onClick={onClickHandler} />
+          <Item val="2" onClick={onClickHandler} />
+          <Item val="3" onClick={onClickHandler} />
+          <Operators val="+" onClick={onClickHandler} onKeyDown={onHandleKeyDown} />
+          <Zero onClick={onClickHandler} />
+          <Item val="." onClick={onClickHandler} onKeyDown={onHandleKeyDown} />
+          <Operators val="=" onClick={onClickHandler} />
+        </div>
+      </div>
+      <Quotes />
+    </>
```

Comment:
- [ ] Instead of here, the `Quote` component should be imported into the main component (`App` component). Therefore you will have something that closely resembles this 👇 
```javascript
const App = () => (
  <>
    <Calculator />
    <Quotes />
  </>
);
```